elastic/kibana
# Add log rate spike analysis to Logs alert details page

Resolves https://github.com/elastic/actionable-observability/issues/8
Partly implements https://github.com/elastic/kibana/issues/159953

Adds log rate spike analysis to Logs threshold alert details page for the "Count" alerts.

The logs spike analysis is shown for the following configurations:
- Count alerts when condition is to check "more than" or "more than or equals" threshold

The logs spike analysis is hidden for the following configurations:
- Ratio alerts
- If rule params contain group by
- If the condition is to check "less than" or "less than or equals" threshold

### Note about Co-Pilot
The PR also integrates Co-Pilot prompt. The prompt message needs to be updated with the correct field values from the analysis results, which will be done in a separate PR.

### Manual testing
1. Create a Log threshold rule
2. Ensure "Count" is selected
3. Ensure "more than" or "more than or equals" is selected for the threshold
4. Enter threshold
5. Save rule
6. Wait for alerts to be generated
7. Go to alert details page
8. Ensure "Explain Log Rate Spikes" is shown

[Guide to generate test data](https://docs.google.com/document/d/1PfPU5mjFx_jWFUkgtaHVy7qPvgBC7g1l4LKe7Jn-qUc)

### Feature flag
The Logs threshold alert details page is behind a feature flag.

To test, add the following in the `kibana.dev.yml`:
 `xpack.observability.unsafe.alertDetails.logs.enabled: true`


<img width="1511" alt="Screenshot 2023-06-12 at 20 18 34" src="https://github.com/elastic/kibana/assets/69037875/dd9391ad-53cd-4484-a11c-090f6a9acdd1">
